### ISSUE TITLE
Split out twisted.python.constants into another package

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7351]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)| @cyli reported|
|-|-|
|Trac ID|trac#7351|
|Type|enhancement|
|Created|2014-05-15 03:15:09Z|
|Branch|https://github.com/twisted/twisted/tree/constantly-7351-2|

We should split out `twisted.python.constants` so that things can depend on it without pulling in all of Twisted.  Particularly, machinist depends on it, and it would be nice to avoid a circular dependency..

Proposal:

1)  move twisted.python.constants into a github twisted/constantly repo (constantly, because Glyph has already registered a domain or some such) - this will me removing any twisted-specific logging or dependency on trial

2) move the corresponding documentation there (??? - not sure about this one - whether we should move or just copy)

3) add it as a requirement for twisted and import it instead

4) eventually deprecate and remove entirely from twisted

<details><summary>Searchable metadata</summary>

```
trac-id__7351 7351
type__enhancement enhancement
reporter__cyli cyli
priority__normal normal
milestone__ 
branch__constantly_7351_2 constantly-7351-2
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1400123709000000 1400123709000000
changetime__1471243826067521 1471243826067521
version__None None
owner__hawkowl hawkowl
cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

It's not a domain, it's [this PyPI distribution identifier](https://warehouse.python.org/project/constantly/).

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)|@hawkowl set owner to @hawkowl|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

Taken an initial stab at https://github.com/hawkowl/constantly - need to sort out py3 tests and such.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

Disregard that - initial PR is here: https://github.com/twisted/constantly/pull/1

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl commented|
|-|-|

(In [46495]) Branching to constantly-7351.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_7"></a>|@hawkowl commented|
|-|-|

This is blocked on all builders becoming virtualenv builders; 12.04 is a pygtk builder and so I need to expand symlinking in the gi stuff for pygtk.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_9"></a>|@glyph commented|
|-|-|

Replying to [hawkowl](#note_7):
> This is blocked on all builders becoming virtualenv builders; 12.04 is a pygtk builder and so I need to expand symlinking in the gi stuff for pygtk.

Is there a ticket for this somewhere?

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_10"></a>|@hawkowl commented|
|-|-|

Yes: https://github.com/twisted-infra/braid/issues/151

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_11"></a>|@hawkowl commented|
|-|-|

(In [47127]) Branching to constantly-7351-2.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_13"></a>|@hawkowl commented|
|-|-|

Happy with this. Please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_14"></a>|@adiroiban commented|
|-|-|

Looks good.

Only minor comments

1. in ``python/__init__.py` why import unicode?
2. Is this deprecatedin 16.0.0 ... or 16.3.0 ?
3. Please fix the pyflakes error. Not sure how we ended with it...but I hope that we can keep pyflakes clean.

please check my comments and merge.

thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_15"></a>|@hawkowl set status to `closed`|
|-|-|

In changeset 3b2d84a28c52dfd16d52846c37ed07e1be41dd77
```
#!CommitTicketReference repository="" revision="3b2d84a28c52dfd16d52846c37ed07e1be41dd77"
Merge constantly-7351-2: Split out twisted.python.constantly into its own package

Author: hawkowl
Reviewers: glyph, adiroiban
Fixes: #7351
```
